source/postprocess/visualization/velocity_residual.cc

```diff
@@ -0,0 +1,169 @@
+	    Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set, dim);
+	  }
+
+
+      template <int dim>
+      void
+      VelocityResidual<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+    	Assert ((computed_quantities[0].size() == dim), ExcInternalError());
+        auto cell = input_data.template get_cell<DoFHandler<dim> >();
+
+        const double velocity_scaling_factor =
+                      this->convert_output_to_years() ? year_in_seconds : 1.0;
+
+        for (unsigned int q=0; q<computed_quantities.size(); ++q)
+          for (unsigned int d = 0; d < dim; ++d)
+        	computed_quantities[q](d)= 0;
+
+        // We only want to output dynamic topography at the top and bottom
+        // boundary, so only compute it if the current cell has
+        // a face at the top or bottom boundary.
+        bool cell_at_top_boundary = false;
+        for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+          if (cell->at_boundary(f) &&
+              (this->get_geometry_model().translate_id_to_symbol_name (cell->face(f)->boundary_id()) == "top"))
+            cell_at_top_boundary = true;
+
+
+        if (cell_at_top_boundary)
+		{
+		  for (unsigned int q=0; q<computed_quantities.size(); ++q)
+		  {
+			Tensor<1,dim> velocity;
+		    if (use_spherical_unit_vectors)
+			{
+		      for (unsigned int d = 0; d < dim; ++d)
+		    	velocity[d] = Utilities::AsciiDataBoundary<dim>::get_data_component(
+		    	  							this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"),
+		    	  							input_data.evaluation_points[q], d);
+		      velocity = Utilities::Coordinates::spherical_to_cartesian_vector(velocity, input_data.evaluation_points[q]);
+
+		      for (unsigned int d = 0; d < dim; ++d)
+		        computed_quantities[q](d) = (velocity[d] - input_data.solution_values[q][d] * velocity_scaling_factor) ;
+			}
+		    else
+		    {
+		      for (unsigned int d = 0; d < dim; ++d)
+		      {
+			  velocity[d] = Utilities::AsciiDataBoundary<dim>::get_data_component(
+													  this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"),
+													  input_data.evaluation_points[q], d);
+			  computed_quantities[q](d) = (velocity[d] - input_data.solution_values[q][d] * velocity_scaling_factor) ;
+
+		      }
+		    }
+		  }
+		}
+      }
+
+
+      template <int dim>
+      void
+	  VelocityResidual<dim>::declare_parameters (ParameterHandler &prm)
+	  {
+    	prm.enter_subsection("Postprocess");
+    	{
+          prm.enter_subsection("Visualization");
+    	  {
+		    Utilities::AsciiDataBase<dim>::declare_parameters(prm,
+						  "$ASPECT_SOURCE_DIR/data/initial-temperature/adiabatic-boundary/",
+						  "adiabatic_boundary.txt", "Surface properties");
```

Comment:
When you no longer derive from `AsciiDataBase` replace this line with the content of the the function you call here.